src/main/java/net/neoforged/moddevgradle/internal/IntelliJIntegration.java

```diff
@@ -0,0 +1,282 @@
+                        LOG.info("Not creating IntelliJ run {} since its prepare task {} is disabled", run, prepareTask);
+                        continue;
+                    }
+                    addIntelliJRunConfiguration(project, runConfigurations, outputDirectory, run, prepareTask);
+                }
+            }
+        });
+    }
+
+    @Override
+    protected void registerProjectSyncTask(TaskProvider<?> task) {
+        // Since this does not just configure a data model but actually runs an additional task, we only do this
+        // when IntelliJ is actually reloading the Gradle project right now.
+        if (!IdeDetection.isIntelliJSync()) {
+            return;
+        }
+
+        project.afterEvaluate(ignored -> {
+            // Also run the sync task directly as part of the sync. (Thanks Loom).
+            var startParameter = project.getGradle().getStartParameter();
+            var taskRequests = new ArrayList<>(startParameter.getTaskRequests());
+
+            taskRequests.add(new DefaultTaskExecutionRequest(List.of(task.getName())));
+            startParameter.setTaskRequests(taskRequests);
+
+        });
+    }
+
+    @Override
+    public void configureTesting(SetProperty<ModModel> loadedMods,
+                                 Property<ModModel> testedMod,
+                                 Provider<Directory> runArgsDir,
+                                 File gameDirectory,
+                                 Provider<RegularFile> programArgsFile,
+                                 Provider<RegularFile> vmArgsFile) {
+        // Write out a separate file that has IDE specific VM args, which include the definition of the output directories.
+        // For JUnit we have to write this to a separate file due to the Run parameters being shared among all projects.
+        var intellijVmArgsFile = runArgsDir.map(dir -> dir.file("intellijVmArgs.txt"));
+
+        var outputDirectory = IntelliJOutputDirectoryValueSource.getIntellijOutputDirectory(project);
+        var ideSpecificVmArgs = RunUtils.escapeJvmArg(getModFoldersProvider(project, outputDirectory, loadedMods, testedMod).getArgument());
+        try {
+            var vmArgsFilePath = intellijVmArgsFile.get().getAsFile().toPath();
+            Files.createDirectories(vmArgsFilePath.getParent());
+            // JVM args generally expect platform encoding
+            FileUtils.writeStringSafe(vmArgsFilePath, ideSpecificVmArgs, StringUtils.getNativeCharset());
+        } catch (IOException e) {
+            throw new GradleException("Failed to write VM args file for IntelliJ unit tests", e);
+        }
+
+        // Configure IntelliJ default JUnit parameters, which are used when the user configures IJ to run tests natively
+        // IMPORTANT: This affects *all projects*, not just this one. We have to use $MODULE_WORKING_DIR$ to make it work.
+        // TODO: remove getRootProject call here, this is just for testing!
```

Comment:
This is inherited from my neodev-related changes. We need to check where this comes from again. I don't see the `getRootProject` call anymore? I think the point was to make this work when only a subproject would call `configureTesting`. (In this case it was the tests project of course).